tensorflow/lite/micro/kernels/fully_connected_common.cc

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/reference/fully_connected.h"
+#include "tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/fully_connected.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+
+namespace tflite {
+namespace {
+#if !defined(XTENSA)
+FullyConnectedParams ToFloatParams(TfLiteFusedActivation activation) {
+  FullyConnectedParams op_params;
+  CalculateActivationRange(activation, &op_params.float_activation_min,
+                           &op_params.float_activation_max);
+  return op_params;
+}
+#endif  // !defined(XTENSA)
+}  // namespace
+
+const int kFullyConnectedInputTensor = 0;
+const int kFullyConnectedWeightsTensor = 1;
+const int kFullyConnectedBiasTensor = 2;
+const int kFullyConnectedOutputTensor = 0;
+
+void* InitFullyConnectedReference(TfLiteContext* context, const char* buffer,
+                                  size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  return context->AllocatePersistentBuffer(
+      context, sizeof(OpDataFullyConnectedReference));
+}
+
+#if !defined(XTENSA)
```

Comment:
@pnikam-cad @nyadla-sys @kpraving

These ifdefs are very undesirable but currently needed because the xtensa linker does not appear to be dropping unused symbols.

If I build the keyword_benchmark without this define the size is ~2.5 KB larger than if I keep this define. However, all the functions within this define are unused and I would expect the linker to be able to drop those symbols. 

I will give more details tomorrow but wanted to flag this issue.

Reply:
this was user-error on my part -- I very likely missed a `make clean`.

Filed https://github.com/tensorflow/tensorflow/issues/46261 to fix the underlying cause.